chain/chain/src/resharding.rs

```diff
@@ -517,4 +574,77 @@ impl Chain {
         }
         Ok(())
     }
+
+    /// Preprocessing stage for on demand resharding. This method should be called only by the resharding tool.
+    pub fn custom_build_state_for_resharding_preprocessing(
+        &self,
+        // The resharding will execute on the post state of the target_hash block.
+        target_hash: &CryptoHash,
+        // The block hash of the state snapshot.
+        snapshot_hash: &CryptoHash,
```

Comment:
I think you no longer rely on the state snapshot, do you? This was originally tailored for the approach Polina was taking. Can you update the comment and / or variable name?

Reply:
Yes this variable is not needed anymore